### DISCUSSION TITLE
How do I insert a React component into the middle of a string?

### DISCUSSION THREAD

**Discussion by liegeandlief** [ivanhofer/typesafe-i18n#305] [Q&A]
Given that I have a string as follows:

> By continuing, you agree to our **Privacy Policy** and consent to receive marketing messages

Where `Privacy Policy` is a link, how would I use this library to translate and output this?

I am using React and would need to wrap `Privacy Policy` in a custom `Link` component.

**Comment by ivanhofer** [comment]
Hi @liegeandlief, you could try it with a soultion like this: https://github.com/ivanhofer/typesafe-i18n#how-do-i-render-a-component-inside-a-translation

**Comment by liegeandlief** [comment]
Thanks @ivanhofer. I ended up converting the string to Markdown using https://github.com/probablyup/markdown-to-jsx to handle rendering the correct `Link` component.

**Reply by ivanhofer** [reply]
@liegeandlief yes that's probably the better solution for simple cases 👍
